### ISSUE TITLE
I can't reproduce the performance as paper report

### ISSUE THREAD

**Issue by lixiangpengcs** [stanfordnlp/mac-network#11]
Thanks for your interesting work and I met some confusions when I run your code.
I run mac-network on GQA dataset as your github guidance. But the valid accuracy is lower than the paper mentioned. I run your code and the performance is listed as:
mac-network valid accuracy: 43.82
GQA-LSTM valid accuracy: 45.54
GQA-LSTMCNN valid accuracy: 42.03
Am I doing something wrong?

**Comment by dorarad** [comment]
Hi, thanks a lot for pointing that out! 
It looks like something is wrong but probably on my side (btw for how many epochs have you trained the networks?) The numbers reported in the paper are the correct ones and were produced with a local version of the code, so it looks like there's some mismatch between my local version and the public version on github - I will look into it today and upload a fix as soon as possible!

**Comment by dorarad** [comment]
alright I think I solved it there was some typo in the preprocess.py file it looked at "idx" field instead of "index" which led to the model looking at the wrong images. It's fixed now so should be working (I'm running an experiment to validate it).

**Comment by cengzy14** [comment]
Hi! What results should mac-network get in validation set? I fixed the typo error, but the accuracy was still quite low.

**Comment by dorarad** [comment]
Hi! validation performance (on validation set)
CNN: 18.76
LSTM: 44.21
CNN+LSTM: 49.24
bottomUp: 52.16
MAC: 57.5 
Performance on the newer test-dev set (which I recommend for validation/development purposes) are a bit lower, about a couple of percents, and match the models' scores on the test set, reported in the paper.

You may need to delete temporary files to get rid of the problem `--generatedPrefix "newfeatures" --expName "newexp`, and if this still doesn't work there may be a need to rerun the `merge.py` commands in the readme. Would be great if you could please post the command/s you have tried

**Comment by cengzy14** [comment]
Thx for your timely reply! BTW, what is the temporary file, and can I validate the accuracy on the test-dev set by myself?

**Comment by dorarad** [comment]
sure! test-dev is available for download in the new data file on the readme page and also in the gqa download page here [visualreasoning.net/download.html](visualreasoning.net/download.html).
Before the model runs experiments it generates some temporary files of preprocessed questions, vocabularies etc. that it later uses for training.

**Comment by cengzy14** [comment]
Thx! I noticed that there may be some mistakes in https://evalai.cloudcv.org/web/challenges/challenge-page/225/submission. When I choose dev 2019, it shows "Total remaining submissions: 5", which is confusing.

**Comment by dorarad** [comment]
np! I think the settings there got changed it should have allowed more for sure I'll make sure to fix it!
